src/serde/de.rs

```diff
@@ -401,16 +401,21 @@ impl<'a, 'de> de::Deserializer<'de> for ValueDeserializer<'a, 'de> {
         use IonType::*;
         match self.value.ion_type() {
             String => visitor.visit_enum(UnitVariantAccess::new(self)),
-            Struct => visitor.visit_enum(VariantAccess::new(self)),
-            _ => IonResult::decoding_error("expected a string or struct enum representation"),
+            // All the parameterized Rust enums uses annotations for representing enum variant
+            _ => visitor.visit_enum(VariantAccess::new(self)),
         }
     }
 
     fn deserialize_identifier<V>(self, visitor: V) -> Result<V::Value, Self::Error>
     where
         V: Visitor<'de>,
     {
-        visitor.visit_str(self.value.read()?.expect_text()?)
+        // annotations are currently only supported with parameterized Rust enums
```

Comment:
What exactly does this mean? Does the rustdoc for this feature need to be updated as well?

Reply:
This means that a serialized Ion data will only contain annotation in the case of parameterized Rust enums. And yes, I did add more comments in the rustdocs for it updating the table of serde data type to Ion type mappings: https://github.com/amazon-ion/ion-rust/pull/733/files#diff-de73824c75ee6203daf57ccbc5aee2ab3e2b46fbd9219d494c190298a7a8e745R43

Reply:
Oh, I saw that but didn't make the connection. It might be worth renaming the table to say "Ion representation" instead of "Ion data type" now that we're including more information than just the data type.

Reply:
Sure. I will change that.